sda-commons-server-jackson/src/test/java/org/sdase/commons/server/jackson/errors/JerseyValidationExceptionMapperTest.java

```diff
@@ -0,0 +1,66 @@
+package org.sdase.commons.server.jackson.errors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.google.common.base.CaseFormat;
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import java.util.stream.Stream;
+import javax.validation.constraints.NotNull;
+import org.hibernate.validator.constraints.ISBN;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class JerseyValidationExceptionMapperTest {
+
+  @ParameterizedTest
+  @MethodSource("testData")
+  void shouldConvertToSnakeCase(String given, String expected) {
+    String actual = JerseyValidationExceptionMapper.camelToUpperSnakeCase(given);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @ParameterizedTest
+  @MethodSource("realTestData")
+  void differenceToDefault(String given) {
+    String actual = JerseyValidationExceptionMapper.camelToUpperSnakeCase(given);
+    String expected = new PropertyNamingStrategies.UpperSnakeCaseStrategy().translate(given);
+    // would fail on: EAN, ISBN, URL, CNPJ, CPF, NIP, PESEL, REGON, INN
+    assumeThat(actual).isEqualTo(expected);
+  }
+
+  @ParameterizedTest
+  @MethodSource("realTestData")
+  void shouldConvertToSnakeCaseLikeGuava(String given) {
+    String actual = JerseyValidationExceptionMapper.camelToUpperSnakeCase(given);
+    String expectedGuava = CaseFormat.UPPER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, given);
+    assertThat(actual).isEqualTo(expectedGuava);
+  }
+
+  static Stream<Arguments> testData() {
+    return Stream.of(
+        Arguments.of("NotNull", "NOT_NULL"),
+        Arguments.of("Email", "EMAIL"),
+        Arguments.of("ISBN", "I_S_B_N"), // how Guava did it, default Jackson: "ISBN"
```

Comment:
I really find Guava's behaviour contra-intuitive. Thanks for adding the tests and the comments. Would be nice to use the plain `UpperSnakeCaseStrategy` in v3.